src/k8s-configuration/azext_k8s_configuration/_validators.py

```diff
@@ -44,3 +58,89 @@ def __validate_k8s_name(param_value, param_name, max_len):
+        if not https_auth_set and scheme == 'https':
+            logger.warning('Warning! https url is being used without https auth params, ensure the repository '
+                           'url provided is not a private repo')
+    else:
+        if https_auth_set:
+            raise MutuallyExclusiveArgumentError(
+                'Error! https auth (--https-user and --https-key) cannot be used with a non-http(s) url',
+                'Verify the url provided is a valid http(s) url and not an ssh url')
+
+
+def _validate_known_hosts(knownhost_data):
+    try:
+        knownhost_str = _from_base64(knownhost_data).decode('utf-8')
+    except Exception as ex:
+        raise InvalidArgumentValueError(
+            'Error! ssh known_hosts is not a valid utf-8 base64 encoded string',
+            'Verify that the string provided safely decodes into a valid utf-8 format') from ex
+    lines = knownhost_str.split('\n')
+    for line in lines:
+        line = line.strip(' ')
+        line_len = len(line)
+        if (line_len == 0) or (line[0] == "#"):
+            continue
+        try:
+            host_key = HostKeyEntry.from_line(line)
+            if not host_key:
+                raise Exception('not enough fields found in known_hosts line')
+        except Exception as ex:
+            raise InvalidArgumentValueError(
+                'Error! ssh known_hosts provided in wrong format',
+                'Verify that all lines in the known_hosts contents are provided in a valid sshd(8) format') from ex
+
+
+def _validate_private_key(ssh_private_key_data):
+    invalid_rsa_key, invalid_ecc_key, invalid_dsa_key, invalid_ed25519_key = (False, False, False, False)
+    try:
+        RSA.import_key(_from_base64(ssh_private_key_data))
+    except ValueError:
+        invalid_rsa_key = True
+        try:
+            ECC.import_key(_from_base64(ssh_private_key_data))
+        except ValueError:
+            invalid_ecc_key = True
+            try:
+                DSA.import_key(_from_base64(ssh_private_key_data))
+            except ValueError:
+                invalid_dsa_key = True
+                try:
+                    key_obj = io.StringIO(_from_base64(ssh_private_key_data).decode('utf-8'))
+                    Ed25519Key(file_obj=key_obj)
+                except SSHException:
+                    invalid_ed25519_key = True
+
+    if invalid_rsa_key and invalid_ecc_key and invalid_dsa_key and invalid_ed25519_key:
+        raise InvalidArgumentValueError(
+            'Error! --ssh-private-key provided in invalid format',
+            'Verify the key provided is a valid PEM-formatted key of type RSA, ECC, DSA, or Ed25519')
+
+
+# pylint: disable=broad-except
+def _validate_cc_registration(cmd):
+    try:
+        rp_client = _resource_providers_client(cmd.cli_ctx)
+        registration_state = rp_client.get(consts.PROVIDER_NAMESPACE).registration_state
+
+        if registration_state != "Registered":
```

Comment:
It probably is the same mixed case always; but wondering if it needs to be a case insensitive comparison.  Also, make this a const?